### DISCUSSION TITLE
Upgrading to version v0.5

### DISCUSSION THREAD

**Discussion by evertramos** [evertramos/nginx-proxy-automation#267] [Q&A] [answered]
Here we will discuss error related to upgrading. Before you start a subject please read it carefully the [upgrade guide](https://github.com/evertramos/nginx-proxy-automation/blob/master/docs/upgrade-guide.md).

**Comment by shoodle** [comment]
Thank you for the new version. Following the "upgrade-guide" there were some glitches I am describing below.

2.4 bascript submodule: Delete the line "cd basescript", because the git commands have to be entered in the main directory.

3. run fresh-start.sh: I did not expect to get into an interactive session with questions about the various settings for .env.

The script also sets the data directory to a wrong path: <directory path>/bin/../data

I kept the old proxy name and in the end the process was successful.

**Reply by evertramos** [reply]
Hello @shoodle,

Thank you for your message!

1. `cd basescript` command

In fact, you are right, we don´t need to be inside the basescript folder... it works that way as well, but better keep it simple. I have already updated, thanks!

2. `fresh-start.sh` script interaction

Yes, the old way was manually done. I like to automate everything that will be repeated somehow. There were an [_'issue'_](https://github.com/evertramos/nginx-proxy-automation/issues/259) related to it... 

Sorry that it was not what you expected, but even though you can skip all those interactions and messages, the way I use in fact at the end of the day. There is a minimal arguments that you must add in order to keep it on track as you wish. Check the helper (`$ ./fresh-start.sh -h`). But for now you would use the follow:

```bash
$ ./fresh-start.sh --yes --silent --skip-docker-image-check -d [YOUR_FULL_PATH_TO_NGINX_DATA] -e [YOUR_EMAIL_ADDRESS]
```

And with the line above it will be really straight forward to fire the new proxy, which is a lot less typing the the previous version, which you should look into the _.env_ file to update all required options.

I really hope you enjoy this!

3. Data folder location

This is not _wrong_ in fact... once, as suggested by a user ([here](https://github.com/evertramos/nginx-proxy-automation/pull/108)).

So, always set the _'./data'_ path where you want to place your files... or we will keep it in the same folder where you cloned the project.

Hope that explains, any further points or suggestion please let me know.

**Comment by evertramos** [comment, marked as answer]
:warning: 

The upgrade guide was so nice that I have done a few updates in different servers and contexts, not only my servers, that I would update myself for any _patreon supporters_.

:heart: 

If you with to do it yourself, which is pretty easy, this is the guide:

https://github.com/evertramos/nginx-proxy-automation/blob/master/docs/upgrade-guide.md

:+1:
